Documentation/contributors/design-proposals/operator-validation-library.md

```diff
@@ -0,0 +1,298 @@
+
+```bash
+$ export PATH=$PATH:$(go env GOPATH)/bin
+```
+
+This adds your workspace's bin subdirectory to your PATH. As a result, you can use the `operator-verify` tool anywhere on your system. Otherwise, you would have to `cd` to your workspace's `bin` directory to run the executable. Verify that we can find `operator-verify` in our new `PATH`:
+
+```bash
+$ which operator-verify
+```
+
+This should return something like:
+
+```
+~/go/bin/operator-verify
+```
+
+To verify that the library installed correctly, use it to validate the ClusterServiceVersion yaml,
+
+```
+$ operator-verify csv /path/to/filename.yaml
+```
+
+## Usage
+
+The command line tool's `help` flag gives the following output:
+
+```text
+operator-verify is a command line tool for the Operator Manifest Verification Library. This library provides functions to validate the operator manifest bundles against Operator-Lifecycle-Manager's clusterServiceVersion type, customResourceDefinitions, and package yamls. Currently, this application supports static validation of both individual operator files and the manifest bundle. 
+
+Usage:
+  operator-verify [flags]
+  operator-verify [command]
+
+Available Commands:
+  help        Help about any command
+  csv         Validate CSV against OLM's type
+  crd         Validate manifest CRDs
+  package     Validate package yaml
+  manifest    Validate operator manifest bundle  
+
+Flags:
+  -h, --help    help for operator-verify
+  -i, --ignore  ignore warnings in log
+
+Use "operator-verify [command] --help" for more information about a command.
+```
+
+
+### Commands 
+**For individual files**: 
+
+```
+$ operator-verify csv /path/to/csv.yaml
+```
+
+Validates the given CSV yaml file against OLM type and reports errors and warnings as described in the `List of Errors` section below.
+
+Similarly, 
+
+```
+$ operator-verify crd /path/to/crd.yaml
+```
+
+and 
+
+```
+$ operator-verify package /path/to/package.yaml
+```
+
+validates CRD and package yaml, respectively. The `crd` command can accept and validate both a single yaml file and a bunch of CRDs at once and report a separate log for each.
+
+**For manifest bundle**:
+
+```
+$ operator-verify manifest /path/to/bundle
+```
+
+Here `path/to/bundle` is a directory structure as per the [operator manifest format](https://github.com/operator-framework/operator-registry#manifest-format). This command reports errors and/or warnings for each file in the bundle.
+
+Using the `help` flag with `manifest` command we get,
+
+```text
+Validates the manifest bundle as a whole, in addition to validating individual operator files. `manifest` reports errors/warnings for each file in the bundle. It works both with a directory structure (as per operator manifest format) and an operator image. `manifest` can also validate only the CSVs or CRDs present in a bundle. See flags for more information. 
+
+Usage:
+  operator-verify verify [flags]
```

Comment:
@dweepgogia I think this should be `operator-verify manifest [flags]` no?

Reply:
Yes, you're correct! Good catch, thanks!